### PR TITLE
cpu: samd21: fix gpio irq handling compile handling

### PR DESCRIPTION
**Pull request by kaspar030**
fixes compile issue exposed in #2852 (bug shows when compiling with -Os)

(this gpio implementation needs a complete rewrite)

### CODE REVIEW

**Comment by OlegHahm** [comment]
This fixes an issue by an PR that's not even merged?

**Comment by kaspar030** [comment]
@OlegHahm The PR exposed a bug.

**Comment by haukepetersen** [comment]
this PR does suppress the warning, but does not fix the issue:
say `GPIO_NUMOF` is 8. Then the `sizeof(gpio_config)` is 8. Now your loop accesses this array from 0 to 16 -> this seems off :-)

**Comment by bapclenet** [comment]
Wait, if we set ``GPIO_NUMOF`` to 1 and we use only ``PA04`` as an int for ``GPIO1``, ``extint`` for ``PA04`` is 4 so ``gpio_config[extint].cb = cb;`` will crash because of ``static gpio_state_t gpio_config[GPIO_NUMOF];``

**Comment by kaspar030** [comment]
@OlegHahm Could you give this a try?

File: cpu/samd21/periph/gpio.c [hunk 1]
```diff
@@ -36,7 +36,7 @@ typedef struct {
     void *arg;          /**< argument passed to the callback */
 } gpio_state_t;
 
-static gpio_state_t gpio_config[GPIO_NUMOF];
+static gpio_state_t gpio_config[16];
```
Comment (OlegHahm) [inline cpu/samd21/periph/gpio.c]: please replace with a non-literal

**Comment by OlegHahm** [comment]
This PR alone seems to work - will test with #2852.

**Comment by OlegHahm** [comment]
ACK, please squash.

**Comment by haukepetersen** [comment]
ACK

**Comment by PeterKietzmann** [comment]
Third ACK. Please squash  :-)

**Comment by riot-labs** [comment]
squashed, rebased

**Comment by kaspar030** [comment]
squashed, rebased

**Comment by PeterKietzmann** [comment]
And go

**Pull request merged**